lib/seahorn/BvOpSem2.cc

```diff
@@ -553,6 +555,32 @@ class OpSemVisitor : public InstVisitor<OpSemVisitor>, OpSemVisitorBase {
     setValue(*CS.getInstruction(), res);
   }
 
+  void visitAssertIf(CallSite CS) {
+    Expr ante = lookup(*CS.getArgument(0));
+    Expr conseq = lookup(*CS.getArgument(1));
+    m_ctx.m_z3_solver->reset();
+    m_ctx.m_z3_solver->assertExpr(ante);
+    boost::tribool res = m_ctx.m_z3_solver->solveAssuming(m_ctx.side());
+    if (res == boost::tribool::true_value) {
+      LOG("opsem", INFO << "Antecedent is sat: " << *CS.getInstruction() << "\n");
+    } else if (res == boost::tribool::false_value) {
+      LOG("opsem", ERR << "Antecedent is unsat: " << *CS.getInstruction() << "\n");
+    } else if (res == boost::tribool::indeterminate_value) {
+      LOG("opsem", WARN << "Antecedent is indeterminate: " << *CS.getInstruction() << "\n");
+    }
+    m_ctx.m_z3_solver->reset();
+    m_ctx.m_z3_solver->assertExpr(conseq);
+    res = m_ctx.m_z3_solver->solveAssuming(m_ctx.side());
+    if (res == boost::tribool::true_value) {
+      LOG("opsem", INFO << "Consequent is sat: " << *CS.getInstruction() << "\n");
+    } else if (res == boost::tribool::false_value) {
+      LOG("opsem", ERR << "Consequent is unsat: " << *CS.getInstruction() << "\n");
+    } else if (res == boost::tribool::indeterminate_value) {
+      LOG("opsem", WARN << "Consequent is indeterminate: " << *CS.getInstruction() << "\n");
+    }
+    setValue(*CS.getInstruction(), m_ctx.alu().getTrue());
```

Comment:
`assert.if` should be `void`. There is no return value.

Reply:
Fixed.